weavertest/multi.go

```diff
@@ -40,13 +42,14 @@ const matchNothingRE = "a^" // Regular expression that never matches
 // component level configs. config is allowed to be empty.
 //
 // Future extension: allow options so the user can control collocation/replication/etc.
-func initMultiProcess(ctx context.Context, name string, isBench bool, runner Runner, logWriter func(*protos.LogEntry)) (context.Context, func() error, error) {
+func initMultiProcess(ctx context.Context, t testing.TB, isBench bool, runner Runner, logWriter func(*protos.LogEntry)) (context.Context, func() error, error) {
```

Comment:
That's a good point. I think that I am leaning towards sticking with letting weaver/weavertest call Main.Main (as opposed to the user explicitly calling it) to keep things similar between non-test and test code. Otherwise, the user has to remember that in scenario Main.Main is called automatically and in another they are responsible.

Note also that Main.Main will only run in weavelets that are hosting weaver.Main. So this could just be considered part of the issue of how test code controls replication.

Reply:
Sounds good to me!

Reply:
I was writing some unit tests and ran into two things that are a bit annoying with the new API (assuming there is more than one test). They are both related to `Main` being run more than once.

1. You can't call `http.HandleFunc` (or any other function that registers an endpoint with the default mux). When the second test runs, it calls `Main` again, and `http.HandleFunc` panics. 
2. If your server listens on a fixed port, like `localhost:9000`, the second test will fail with a `bind: address already in use` if your `Main` doesn't respect the context it receives.

I'm not sure how to fix these issues, but wanted to document them somewhere.